src/main/java/net/sf/jabref/gui/keyboard/KeyBindingsDialogViewModel.java

```diff
@@ -0,0 +1,210 @@
+import javafx.scene.control.SelectionMode;
+import javafx.scene.control.TreeItem;
+import javafx.scene.control.TreeTableColumn;
+import javafx.scene.control.TreeTableView;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.KeyEvent;
+import javafx.stage.Stage;
+
+public class KeyBindingsDialogViewModel {
+
+    private KeyBindingRepository keyBindingRepository;
+
+    @FXML
+    private TreeTableView<KeyBindingWrapper> keyBindingsTable;
+
+    @FXML
+    private TreeTableColumn<KeyBindingWrapper, String> actionColumn;
+
+    @FXML
+    private TreeTableColumn<KeyBindingWrapper, String> shortcutColumn;
+
+    @FXML
+    private Button closeButton;
+
+    @FXML
+    private Button resetButton;
+
+
+    @FXML
+    private void initialize() {
+        keyBindingsTable.getSelectionModel().setSelectionMode(SelectionMode.SINGLE);
+        ButtonBar.setButtonData(resetButton, ButtonData.LEFT);
+    }
+
+    public void initializeView() {
+        registerKeyEvents();
+        populateTable();
+        bindColumnValues();
+    }
+
+    /**
+     * Read all keybindings from the keybinding repository and create table keybinding
+     * models for them
+     */
+    private void populateTable() {
+        TreeItem<KeyBindingWrapper> root = new TreeItem<>(new KeyBindingWrapper(null, "", KeyBindingCategory.FILE));
+        for (KeyBindingCategory category : KeyBindingCategory.values()) {
+            TreeItem<KeyBindingWrapper> categoryItem = new TreeItem<>(new KeyBindingWrapper(null, "", category));
+            keyBindingRepository.getKeyBindings().forEach((keyBinding, bind) -> {
+                if (keyBinding.getCategory() == category) {
+                    KeyBindingWrapper keyBindWrapper = new KeyBindingWrapper(keyBinding, bind,
+                            keyBinding.getCategory());
+                    TreeItem<KeyBindingWrapper> keyBindTreeItem = new TreeItem<>(keyBindWrapper);
+                    categoryItem.getChildren().add(keyBindTreeItem);
+                }
+            });
+            categoryItem.setExpanded(true);
+            root.getChildren().add(categoryItem);
+        }
+        root.setExpanded(true);
+        keyBindingsTable.setRoot(root);
+    }
+
+    private void bindColumnValues() {
+        actionColumn
+                .setCellValueFactory(cellData -> new SimpleStringProperty(cellData.getValue().getValue().getName()));
+        shortcutColumn.setCellValueFactory(cellData -> cellData.getValue().getValue().bindProperty());
+    }
+
+    private void registerKeyEvents() {
+        keyBindingsTable.setOnKeyPressed(evt -> grabKey(evt));
+    }
+
+    private Stage getDialogStage() {
+        return (Stage) closeButton.getScene().getWindow();
+    }
+
+    @FXML
+    private void closeDialog() {
+        Stage stage = getDialogStage();
+        stage.close();
+    }
+
+    @FXML
+    private void saveKeyBindingsAndCloseDialog() {
+        Globals.getKeyPrefs().setNewKeyBindings(keyBindingRepository.getKeyBindings());
```

Comment:
Pass the KeyPreference class to this viewmodel instead of the `keyBindingRepository` so that you don't need to access the Globals object here (we want to get rid of it in the end).